api/types.go

```diff
@@ -12,33 +16,63 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
-	IssuerUrl    *string
-	ClientId     *string
-	ClientSecret *string
-	Audience     *string
-	Enabled      *bool
+	Enabled    *bool
+	Filename   *string
+	Attributes OIDCAttributes
+}
+
+type OIDCAttributes struct {
+	IssuerURL    string `yaml:"issuer_url"`
+	ClientID     string `yaml:"client_id"`
+	ClientSecret string `yaml:"client_secret"`
+	Audience     string `yaml:"audience"`
 }
 
 type TokenVerificationConfig struct {
 	Enabled *bool
 	Url     *string
 }
 
-func (c *OIDCConfig) Validate() {
+func (c *OIDCConfig) Read() error {
+	configFile, err := os.Open(*c.Filename)
+	if err != nil {
+		return err
+	}
+
+	data, err := io.ReadAll(configFile)
+	if err != nil {
+		return err
+	}
+
+	err = yaml.Unmarshal(data, c)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *OIDCConfig) Validate() error {
```

Comment:
I see you separated ReadAndValidate on different methods Read and Validate.
:+1: . Aside from separating the concerns one positive aspect we gain from this is that it makes the `Validate` method much easier to unit test as it does not involve file reading logic, in case we want to do it, which we should in future PRs.
Read is still not testing friendly as I mentioned in a previous comment but we can tackle that in a future PR.